### ISSUE TITLE
Base pipe feature released in R 4.1

### ISSUE THREAD

**Issue by ttimbers** [UBC-DSCI/introduction-to-datascience#153]
@trevorcampbell @leem44 - a new base pipe feature was just released in R 4.1. I think we have to think about incorporating this in the book. We don't have to make this decision today, but we should think carefully before we go to print. This is something that could easily date our book really fast if the community adopts it and we do not...

- https://twitter.com/thomas_mock/status/1394693648143499269
- https://stat.ethz.ch/R-manual/R-devel/doc/html/NEWS.html (search for pipe)

**Comment by ttimbers** [comment]
I had a very helpful conversation with @jennybc about this, which I will attempt to summarize below:

Jenny said that the Tidyverse has (long) had an [official policy of supporting past R versions](https://www.tidyverse.org/blog/2019/04/r-version-support/), and what that means for them is that they cannot switch completely over to the new pipe for many years. She thinks this means that there will be some awkward period where this has to be addressed in books and documentation (I think the publishing of our book will fall squarely in that time period...).

She also said that luckily, `magrittr` (where the Tidyverse pipe comes from) was recently updated to make it more like the built-in pipe, so that lots of code will work with either. I think that we do need to be cautious with the book if we adopt the built-in pipe as she also said that a built-in pipe can be implemented in a different way than is possible with an add-on package. 

In summary, she recommended (and I wholeheartedly agree) that we should move to the new, forward-looking syntax (the new built-in pipe). However, we should introduce both (because they will see both in the docs, and in any place they search for help online), and then move to just using the new, built-in one. The one other place where we might want to again discuss both are if we have any examples in the book where the  behaviour would differ between the two pipes. There we would be well served to call attention to these differences.

Thanks again @jennybc for your help and advice here!

**Comment by trevorcampbell** [comment]
+1 to switching to the new pipe but introduce both and tell readers about the difference

**Comment by jennybc** [comment]
Blog posts related to magrittr pipe vs base R pipe:

* <https://www.tidyverse.org/blog/2020/08/magrittr-2-0/> planning, asking for feedback
* <https://www.tidyverse.org/blog/2020/11/magrittr-2-0-is-here/> what actually happened

**Comment by ttimbers** [comment]
Thanks for these refs @jennybc!

**Comment by trevorcampbell** [comment]
I believe this is handled now. Closing (but reopen if I'm wrong!)
